frontend/src/components/dashboard/Analysis.tsx

```diff
@@ -364,7 +435,10 @@ const Analysis = ({
               }
               markAsHandled(evt.native);
 
-              setSplits((splits) => splits.filter((s) => s !== split));
+              dispatch({
+                type: AnalyzerStateActionType.setSplits,
+                splits: (splits) => splits.filter((s) => s !== split),
+              });
```

Comment:
This could be dispatched as a `removeSplit` and correspondingly `addSplit` below, making the logic testable exclusively within the dispatch.